### ISSUE TITLE
ptype2 howto shouldn't advise `stop_incompatible_type()`

### ISSUE THREAD

**Issue by lionel-** [r-lib/vctrs#1731]
In https://vctrs.r-lib.org/reference/howto-faq-coercion.html

Should recommend `vec_default_ptype2()` instead.

**Comment by DavisVaughan** [comment]
Does this include `stop_incompatible_cast()`? Because I feel like being able to call that with custom `details` is pretty useful
https://github.com/DavisVaughan/int64/blob/0922226847a3eec5e4b5887490c8c82b8751a0e5/R/cast.R#L36-L41

**Comment by lionel-** [comment]
yes I think so. The default method is where we implement all the special behaviour. Maybe we could pass `details` to `stop_incompatible_cast()`?

**Comment by DavisVaughan** [comment]
Why do I want the special behavior though? I'm confident that I want a cast/type error and was trying to throw a vctrs compatible one

**Comment by lionel-** [comment]
yes it can indeed actually work if someone overrides the method. For this to work, coercion methods should _return_ the result of the default method instead of failing.

**Comment by DavisVaughan** [comment]
For future us:

> someone overrides the method

is referring to utilizing a restart, like https://github.com/r-lib/vctrs/pull/1719
